tracing-subscriber/src/registry/extensions.rs

```diff
@@ -0,0 +1,224 @@
+// taken from https://github.com/hyperium/http/blob/master/src/extensions.rs.
+
+use crate::sync::{RwLockReadGuard, RwLockWriteGuard};
+use std::{
+    any::{Any, TypeId},
+    collections::HashMap,
+    fmt,
+    hash::{BuildHasherDefault, Hasher},
+};
+
+#[allow(warnings)]
+type AnyMap = HashMap<TypeId, Box<dyn Any + Send + Sync>, BuildHasherDefault<IdHasher>>;
+
+/// With TypeIds as keys, there's no need to hash them. They are already hashes
+/// themselves, coming from the compiler. The IdHasher holds the u64 of
+/// the TypeId, and then returns it, instead of doing any bit fiddling.
+#[derive(Default, Debug)]
+pub struct IdHasher(u64);
+
+impl Hasher for IdHasher {
+    fn write(&mut self, _: &[u8]) {
+        unreachable!("TypeId calls write_u64");
+    }
+
+    #[inline]
+    fn write_u64(&mut self, id: u64) {
+        self.0 = id;
+    }
+
+    #[inline]
+    fn finish(&self) -> u64 {
+        self.0
+    }
+}
+
+/// An immutable, read-only reference to a Span's extensions.
+#[derive(Debug)]
+pub struct Extensions<'a> {
+    inner: RwLockReadGuard<'a, ExtensionsInner>,
+}
+
+impl<'a> Extensions<'a> {
+    pub(crate) fn new(inner: RwLockReadGuard<'a, ExtensionsInner>) -> Self {
+        Self { inner }
+    }
+
+    /// Get a mutable reference to a type previously inserted on this `Extensions`.
```

Comment:
this gets an _immutable_ reference:
```suggestion
    /// Get a reference to a type previously inserted on this `Extensions`.
```
or maybe
```suggestion
    /// Immutably borrows a type previously inserted into this `Extensions`.
```